getting started/democratic dao.sol

```diff
@@ -1,92 +1,106 @@
+        function newProposal(address _recipient, uint _amount, bytes32 _data, string _description) returns (uint proposalID) {
+            if (voterShare.coinBalanceOf(msg.sender)>0) {
+                proposalID = proposals.length++;
+                Proposal p = proposals[proposalID];
+                p.recipient = _recipient;
+                p.amount = _amount;
+                p.data = _data;
+                p.description = _description;
+                p.creationDate = now;
+                p.active = true;
+                ProposalAdded(proposalID, _recipient, _amount, _data, _description);
+                numProposals = proposalID+1;
+            } else {
+                return 0;
             }
-        } else {
-            return 0;
         }
-    }
-    
-    function executeProposal(uint _proposalID) returns (int result) {
-        Proposal p = proposals[_proposalID];
-        /* Check if debating period is over */
-        if (now > (p.creationDate + debatingPeriod) && p.active){   
-            
-            /* tally the votes */
-            for (uint i = 0; i <=  p.votes.length; i++) {
-                Vote v = p.votes[i];
-                int voteWeight = int(voterShare.balances(v.voter)); 
-                p.quorum += uint(voteWeight);
-                result += voteWeight * v.position;
+        
+        function vote(uint _proposalID, int _position) returns (uint voteID){
+            LineCounter(83);
+            if (voterShare.coinBalanceOf(msg.sender)>0 && (_position >= -1 || _position <= 1 )) {
+                LineCounter(85);
+                Proposal p = proposals[_proposalID];
+                if (p.voted[msg.sender] != true) {
+                    LineCounter(88);
+                    voteID = p.votes.length++;
+                    Vote v = p.votes[voteID];
+                    v.position = _position;
+                    v.voter = msg.sender;   
+                    p.voted[msg.sender] = true;
+                    Voted(_proposalID,  _position, msg.sender);
+                }
+            } else {
+                LineCounter(97);
+                return 0;
             }
-            /* execute result */
-            if (p.quorum > minimumQuorum && result > 0 ) {
-                p.recipient.call.value(p.amount)(p.data);
-                p.active = false;
-            } else if (p.quorum > minimumQuorum && result < 0) {
-                p.active = false;
+        }
+        
+        function executeProposal(uint _proposalID) returns (int result) {
+            Proposal p = proposals[_proposalID];
+            /* Check if debating period is over */
+            if (now > (p.creationDate + debatingPeriod) && p.active){   
+                
+                /* tally the votes */
+                for (uint i = 0; i <=  p.votes.length; i++) {
+                    Vote v = p.votes[i];
+                    int voteWeight = int(voterShare.coinBalanceOf(v.voter)); 
+                    p.quorum += uint(voteWeight);
```

Comment:
If you want this to be cheaper, don't sum up the quorum in storage but use a local variable.